### PR TITLE
OpcodeDispatcher: Optimize PSHUF{LW, HW, D}! 

### PR DESCRIPTION
**Pull request by Sonicadvance1**
This is way more optimal!

### CODE REVIEW

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h [hunk 1]
```diff
@@ -1203,6 +1206,23 @@ friend class FEXCore::IR::PassManager;
   // Named constant cache for the current block.
   // Different arrays for sizes 1,2,4,8,16,32.
   OrderedNode *CachedNamedVectorConstants[FEXCore::IR::NamedVectorConstant::NAMED_VECTOR_MAX][6]{};
+  struct IndexNamedVectorMapKey {
+    uint32_t Index{};
+    FEXCore::IR::IndexNamedVectorConstant NamedIndexedConstant;
+    uint8_t log2_size_in_bytes{};
+
+    bool operator==(const IndexNamedVectorMapKey& k) const {
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher.h:1214]:
```suggestion
    bool operator==(const IndexNamedVectorMapKey& k) const = default;
    bool operator!=(const IndexNamedVectorMapKey& k) const = default;
```

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h [hunk 2]
```diff
@@ -1203,6 +1206,23 @@ friend class FEXCore::IR::PassManager;
...
+      return Index == k.Index &&
+             NamedIndexedConstant == k.NamedIndexedConstant &&
+             log2_size_in_bytes == k.log2_size_in_bytes;
+    }
+  };
+  struct IndexNamedVectorMapKeyHasher {
+    std::size_t operator()(const IndexNamedVectorMapKey& k) const {
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher.h:1221]:
```suggestion
    std::size_t operator()(const IndexNamedVectorMapKey& k) const noexcept {
```

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h [hunk 3]
```diff
@@ -1203,6 +1206,23 @@ friend class FEXCore::IR::PassManager;
...
+      return XXH3_64bits((void*)&k, sizeof(k));
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher.h:1222]:
```suggestion
      return XXH3_64bits(&k, sizeof(k));
```

Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h]: Resolved

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h (hunk 2 above)
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h]: Resikved

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h (hunk 1 above)
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h]: Resoulved

**Review by lioncash — APPROVED** [review]

**Pull request merged**